tests/test_converters/test_obo_graph_to_fhir.py

```diff
@@ -206,16 +244,43 @@ def test_convert_so(self):
             )
             dl_path = OUTPUT_DIR / "so.owl.obographs.json"
             out = OUTPUT_DIR / f"{filename}.json"
-            cs: CodeSystem = self._load_and_convert(
+            cs: CodeSystem = self._load_and_convert_json(
                 out,
                 dl_path,
                 dl_url=dl_url,
                 code_system_id=filename.replace("CodeSystem-", ""),
                 code_system_url="http://purl.obolibrary.org/obo/so.owl",
                 native_uri_stems=["http://purl.obolibrary.org/obo/SO_"],
             )
-            # TODO: choose a better threshold
+            # todo: choose a better threshold
             self.assertGreater(len(cs.concept), 100)
-            # TODO: choose a property to assert
+            # todo: choose a property to assert
             # prop_uris: List[str] = [p.uri for p in cs.property]
             # self.assertIn("", prop_uris)
+
+    @unittest.skip("TODO: change to an integration test")
+    def test_convert_so_package(self):
+        """Tests specific to Sequence Ontology (SO)."""
+        if DOWNLOAD_TESTS_ON:
+            filename = "CodeSystem-sequence-ontology"
+            dl_url = (
+                "https://github.com/"
+                "HOT-Ecosystem/owl-on-fhir-content/releases/download/2023-01-13/so.owl.obographs.json"
+            )
+            dl_path = OUTPUT_DIR / "so.owl.obographs.json"
+            zip_outpath: str = self._load_and_convert_npm(
+                OUTPUT_DIR,
+                dl_path,
+                dl_url=dl_url,
+                code_system_id=filename.replace("CodeSystem-", ""),
+                code_system_url="http://purl.obolibrary.org/obo/so.owl",
+                native_uri_stems=["http://purl.obolibrary.org/obo/SO_"],
+                manifest_path=INPUT_DIR / "fhir_npm_manifest_so.json",
+            )
+            unzip_path = os.path.join(OUTPUT_DIR, os.path.basename(zip_outpath).replace(".tgz", ""))
+            with tarfile.open(zip_outpath, "r:gz") as tar:
+                tar.extractall(unzip_path, members=tar.getmembers())
```

Comment:
## `tarfile.extractall used without any validation`
@hrshdhgd Months ago you took a stab at [a recommendation](https://github.com/INCATools/ontology-access-kit/pull/511#issuecomment-1497484498). I just got to trying that out now (added ` members=tar.getmembers()`) and it worked. Thanks!

I want to clean up this PR to make it easier for Chris to review, so I'm going to copy/paste our original conversation here, mark this as resolved, and then hide those original comments as duplciates.

### Original discussion
[Harshad wrote](https://github.com/INCATools/ontology-access-kit/pull/511#issuecomment-1497484498):
<details><summary>Details</summary>
<p>

[Joe wrote](https://github.com/INCATools/ontology-access-kit/pull/511#pullrequestreview-1371937790):
> @hrshdhgd Ever seen this before? If not, I can look into a bit more before raising to Chris or someone else.
> 
> * [ ]  tests/test_converters/test_obo_graph_to_fhir.py:282:1: S202 tarfile.extractall used without any validation. Please check and discard dangerous members.
> 
> I understand what it's saying but I'm not sure how to make it happy. I need to extract a zip file for my test. Haven't looked into what it means by "doing validation" first though.
> 
> [Slack thread](https://monarchinitiative.slack.com/archives/C0222P0736F/p1680652762465309)

 - Firstly, could you please add `fhirjson_conf.json` in `tests/input` into gitignore. It gets generated every time tests run and I'm unsure if it should be version controlled or no. This occurs in all branches of oak. I have added it to `.gitignore` for now in the branch I'm working on but I think you're more knowledgeable about this.
 - I think validation **may** involve providing the file names in the tar file as a list in the `members` param for `extractall`. [See this](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall). My guess is it wants to make sure the receiver of the tar file get what they are expecting, nothing more nothing less (as a security measure).
 - I am not a 100% sure about this solution but worth a shot. I'd do it myself but I cannot find the zip file. I'm assuming it is local to you.
 - Make a list variable with all file names and feed it to `extractall()`. Just a shot in the dark. It may work but not sure.

Hope this helps!

</p>
</details> 

[Joe wrote](https://github.com/INCATools/ontology-access-kit/pull/511#issuecomment-1497878247):
<details><summary>Details</summary>
<p>

@hrshdhgd That's a great suggestion; `members`. I'll try it. The zip file gets created by running the test `test_convert_so_package()`. in `test_obo_graph_to_fhir.py`.

As for `fhirjson_conf.json`, I'm not seeing that anywhere. Did you mean `test/input/fhir_config_example.json`? I was considering using that one for a unit test but haven't done so yet; considering removing it. If you are seeing `fhirjson_conf.json`, I can make a PR / commit to `main` that just adds it to the `.gitignore`.

</p>
</details> 


[Harshad wrote](https://github.com/INCATools/ontology-access-kit/pull/511#issuecomment-1497980860) (and Joe thumbs-upped):
<details><summary>Details</summary>
<p>

> As for fhirjson_conf.json, I'm not seeing that anywhere. 

Run `pytest` and you'll see this file is generated.

</p>
</details>